### PR TITLE
Added ability to retry on initialization errors.

### PR DESCRIPTION
**Pull request by Silvenga**
## Breaking change

When an initialization errors occurs, the frontend will attempt to retry automatically. The original ability to retry manually is still supported.

## Proposed change

This pr adds the ability to automatically "retry" loading the frontend on initialization errors. This is to handle transient problems with networking.

This fixes a recurring problem with a tablet display I use. I use a split horizon DNS setup and the tablet hard codes Google DNS servers, which is used for 25% of the DNS requests. So, randomly, the tablet will show the error screen after reloading (and network resolution failing).

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [X] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

The changes should appear as such:

image

**Testing**: When service workers are enabled and activated, attempt to load the frontend with networking offline (in Chrome's networking tab). This should cause an initialization error, prompting the rendering of the `ha-init-page` component. Observed the count down decreasing and that the page reloads after about 60 seconds.

## Checklist

- [ ] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @Silvenga, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#12103).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by Silvenga** [comment]
I'm not 100% sure the best way to test this change (I run my HA instance in a K8s cluster). Is there a way to bring up HA in a Docker container with the local frontend (Docker container, since I develop on Windows)?

_Silvenga feels bad about this._
_Silvenga hopes that the changes are so small that Silvenga can verify using a beta release from CI._

**Comment by bramkragten** [comment]
You can setup a dev env in a docker container: https://developers.home-assistant.io/docs/frontend/development

File: src/layouts/ha-init-page.ts [hunk 1]
```diff
@@ -7,13 +7,20 @@ class HaInitPage extends LitElement {
 
   @state() showProgressIndicator = false;
 
+  @state() retryInSeconds = 60;
+
   private _showProgressIndicatorTimeout;
 
+  private _retryInterval;
```
Comment (bramkragten) [inline src/layouts/ha-init-page.ts:14]:
```suggestion
  private _retryInterval: number;
```

File: src/layouts/ha-init-page.ts [hunk 2]
```diff
@@ -7,13 +7,20 @@ class HaInitPage extends LitElement {
 
   @state() showProgressIndicator = false;
 
+  @state() retryInSeconds = 60;
```
Comment (bramkragten) [inline src/layouts/ha-init-page.ts:10]:
```suggestion
  @state() private _retryInSeconds = 60;
```

**Comment by zsarnett** [comment]
Can you take a look at the reviews here?

**Comment by Silvenga** [comment]
Apologies, this slipped my mind. The dev container breaks on setting up git for some reason, so I haven't been able to manually test it.

File: src/layouts/ha-init-page.ts (hunk 1 above)
Comment (Silvenga) [reply src/layouts/ha-init-page.ts]: This breaks under modern TS - but updated to use explicit types.

File: src/layouts/ha-init-page.ts (hunk 2 above)
Comment (Silvenga) [reply src/layouts/ha-init-page.ts]: Updated both to use private moditiers.

**Review by zsarnett — APPROVED** [review]

**Review by zsarnett — APPROVED** [review]

**Pull request merged**